app/views/planning_applications/validation/tasks/_application_details.html.erb

```diff
@@ -2,76 +2,43 @@
   <h2 class="app-task-list__section">
     Check application details
   </h2>
-  <ul class="app-task-list__items">
-    <li id="draw_red_line_boundary" class="app-task-list__item">
-      <span class="app-task-list__task-name">
-        <%= link_to_unless(@planning_application.boundary_geojson.present? && @planning_application.officer_can_draw_boundary?, "Draw red line boundary", planning_application_validation_sitemap_path(@planning_application), class: "govuk-link") %>
-      </span>
-      <div class="govuk-task-list__status app-task-list__task-tag">
-      <%= render(
-            StatusTags::DrawRedLineBoundaryComponent.new(
-              boundary_geojson: @planning_application.boundary_geojson
-            )
-          ) %>
-      </div>
-    </li>
-    <%= render(
-          TaskListItems::CheckRedLineBoundaryComponent.new(
-            planning_application: @planning_application
-          )
-        ) %>
-    <li id="constraints-validation-tasks" class="app-task-list__item">
-      <span class="app-task-list__task-name">
-        <%= govuk_link_to "Check constraints", planning_application_validation_constraints_path(@planning_application) %>
-      </span>
-      <div class="govuk-task-list__status app-task-list__task-tag">
-      <%= render(
-            StatusTags::ConstraintsComponent.new(
-              planning_application: @planning_application
-            )
-          ) %>
-      </div>
-    </li>
-    <% if @planning_application.validated? %>
-      <li id="check-description" class="app-task-list__item">
-        <span class="app-task-list__task-name">
-          Planning application has already been validated
-        </span>
-      </li>
-    <% else %>
-      <%= render(
-            TaskListItems::DescriptionChange.new(
-              planning_application: @planning_application
-            )
-          ) %>
-    <% end %>
-    <li id="development-type-for-reporting-task" class="app-task-list__item">
-      <span class="app-task-list__task-name">
-        <%= govuk_link_to "Add reporting details", edit_planning_application_validation_reporting_type_path(@planning_application) %>
-      </span>
+  <%= govuk_task_list(id_prefix: "application-details-tasks") do |task_list|
+        task_list.with_item(html_attributes: {id: "draw-red-line-boundary"}) do |item|
+          task = Validation::DrawRedLineBoundaryTask.new(@planning_application)
+          item.with_title(text: task.task_list_link_text, href: task.task_list_link)
+          item.with_status(text: render(task.task_list_status_component))
```

Comment:
I'm not certain about this. On the one hand if we want this to be automatically generated from a list of workflow tasks, then all the task-specific logic needs to be inside the task class; on the other hand, does that mix logic and presentation too much?

If we wanted to do that, then would be possible to move all of the calls to `StatusTags::BaseComponent` into `WorkflowTask` and then just override it in the handful of tasks that need a specific status component rather than the default.